src/neptune/metadata_containers/run.py

```diff
@@ -451,6 +463,35 @@ def _write_initial_attributes(self):
             # upload default sources ONLY if creating a new run
             upload_source_code(source_files=self._source_files, run=self)
 
+    def _track_dependencies(self) -> None:
+        if self._dependencies == "infer":
+            # using pipreqs here
+            file_name = "neptune-tracked-dependencies.txt"
```

Comment:
I would like to see it via uploading an in-memory file instead of using some temp file on disk.